app/assets/javascripts/controllers/ansible_credentials/ansible_credentials_form_controller.js

```diff
@@ -40,6 +36,18 @@ ManageIQ.angular.app.controller('ansibleCredentialsFormController', ['$window',
       vm.modelCopy = angular.copy( vm.credentialModel );
       miqService.sparkleOff();
     }
+
+    var optionsPromiseResolved = optionsPromise
+      .then(getCredentialOptions)
+      .catch(miqService.handleFailure);
+
+    var dataPromiseResolved = dataPromise
```

Comment:
(`$q.all` can handle undefined there just fine, it makes it into a Promise that resolves to undefined, so no issue there)